### DISCUSSION TITLE
Modifying attributes on form submit buttons

### DISCUSSION THREAD

**Discussion by tobyzerner** [unpoly/unpoly#307] [Ideas]
Currently, as far as I can tell, in the context of a form, you can only specify [modifying attributes](https://unpoly.com/a-up-follow) on the `<form>` tag.

I would like to be able to have fine-grained control according to which button is clicked. For example:

```html
<form id="my-form" up-submit>
  <button>Submit in current layer</button>
  <button up-layer="new">Submit in new layer</button>
</form>
```

Using multiple forms is not an option, as I want to be able to target a single form from input elements outside the form:

```html
<input type="checkbox" form="my-form">
```

Does this sound like something that could be considered?

**Comment by pfw** [comment]
As far as I’m aware there isn’t a way to do that automatically, I have done similar things using a ‘compiler’ on the button. In this case you could catch the click on the button and use up.form.submitOptions to pull out the values on the form attributes, update with options set on the button and pass to up.submit. The compiler can be completely standard across all of your forms.

I’ve often found cases that are a little outside the norm but have found them quite straightforward to handle with a compiler and calling the underlying methods the default attributes would cause to be called.

Peter W.

**Reply by tobyzerner** [reply]
Ah, that's clever. Thanks Peter!
